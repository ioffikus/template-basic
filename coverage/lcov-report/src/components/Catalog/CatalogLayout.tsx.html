<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Catalog/CatalogLayout.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Catalog</a> CatalogLayout.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Layout } from 'antd';
import dispatch from 'src/store/dispatch';
import { getCatalogQuery } from 'src/ducks/catalogQuery';
import { getCategoriesMenu } from 'src/ducks/categoriesMenu';
import { getCatalog } from 'src/ducks/catalog';
import { getCategories } from 'src/ducks/categories';
import { getFilter } from 'src/ducks/filter';
import Catalog from 'src/components/Catalog';
import SiderMenu from 'src/components/SiderMenu';
import CategoriesMenu from 'src/components/CategoriesMenu';
import FilterMenu from 'src/components/FilterMenu';
&nbsp;
class CatalogLayout extends React.Component&lt;{}, {}&gt; {
  static fetchData: Store.IFetchDataFn = abortToken =&gt; {
    return dispatch(getCatalogQuery()).then(() =&gt; {
      return Promise.all([
        dispatch(getCategoriesMenu(abortToken)),
        dispatch(getCatalog(abortToken)),
        dispatch(getCategories(abortToken)),
        dispatch(getFilter(abortToken)),
      ]);
    });
  };
&nbsp;
  render() {
    return (
      &lt;Layout&gt;
        &lt;SiderMenu type="categories"&gt;
          &lt;CategoriesMenu /&gt;
        &lt;/SiderMenu&gt;
        &lt;Layout.Content&gt;
          &lt;Catalog /&gt;
        &lt;/Layout.Content&gt;
        &lt;SiderMenu type="filters"&gt;
          &lt;FilterMenu /&gt;
        &lt;/SiderMenu&gt;
      &lt;/Layout&gt;
    );
  }
}
&nbsp;
export default CatalogLayout;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 27 2017 23:45:35 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
