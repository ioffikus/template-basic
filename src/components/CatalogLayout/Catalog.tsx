import React from 'react';
import { Row, Col } from 'antd';
import Breadcrumbs from 'src/components/Breadcrumbs';
import SearchFrom from './SearchForm';
import ProductItems from './ProductItems';

class Catalog extends React.Component<{}, {}> {
  render() {
    return (
      <div className="root app-container">
        <style jsx>{`
          .root :global(.header) {
            margin-bottom: 30px;
          }
        `}</style>
        <Row type="flex" justify="space-between" className="header">
          <Col span={23}>
            <Breadcrumbs for="catalog" />
          </Col>
          <Col span={1}>
            <SearchFrom />
          </Col>
        </Row>
        <ProductItems />
      </div>
    );
  }
}

export default Catalog;
