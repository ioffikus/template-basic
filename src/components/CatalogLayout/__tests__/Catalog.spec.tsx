import React from 'react';
import { shallow } from 'enzyme';
import Catalog from '../Catalog';

describe(null, () => {
  it('render()', () => {
    const wrapper = shallow(<Catalog />);
    expect(wrapper).toMatchSnapshot();
  });
});
