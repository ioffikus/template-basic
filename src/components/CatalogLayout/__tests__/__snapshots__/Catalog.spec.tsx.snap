// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() 1`] = `
<div
  className="jsx-2818191692 root app-container"
>
  <JSXStyle
    css=".root.jsx-2818191692 .header{margin-bottom:30px;}"
    styleId="2818191692"
  />
  <Row
    className="header"
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <Col
      span={23}
    >
      <Connect(Breadcrumbs)
        for="catalog"
      />
    </Col>
    <Col
      span={1}
    >
      <Connect(SearchForm) />
    </Col>
  </Row>
  <Connect(ProductItems) />
</div>
`;
